src/js/media-poster-image.js

```diff
@@ -15,24 +15,36 @@ template.innerHTML = `
       box-sizing: border-box;
     }
 
+    #container {
+      background-repeat: no-repeat;
+      background-position: var(--media-background-position, var(--media-object-position, center));
+      background-size: var(--media-background-size, var(--media-object-fit, contain));
+    }
     img {
       max-width: 100%;
       max-height: 100%;
       min-width: 100%;
       min-height: 100%;
-      background-repeat: no-repeat;
-      background-position: var(--media-background-position, var(--media-object-position, center));
-      background-size: var(--media-background-size, var(--media-object-fit, contain));
       object-fit: var(--media-object-fit, contain);
       object-position: var(--media-object-position, center);
+      opacity: 1;
+    }
+    img.fade {
+      opacity: 0;
+      transition: opacity 0.15s ease;
+    }
+    img.fade.loaded {
+      opacity: 1;
```

Comment:
Can switch to use `img:not([no-fade])` instead and get rid of everything related to the `fade` class to simplify the code and the new additional JS.

Reply:
Additionally, should we fade in only if a placeholder is provided? I'd probably expect it to show up asap unless placeholder (unless/when we expose the `loading` attribute)